src/Auth0.php

```diff
@@ -522,24 +546,25 @@ public function exchange()
             throw new ApiException('Invalid access_token - Retry login.');
         }
 
-        $accessToken = $response['access_token'];
+        $this->setAccessToken($response['access_token']);
 
-        $refreshToken = false;
         if (isset($response['refresh_token'])) {
-            $refreshToken = $response['refresh_token'];
+            $this->setRefreshToken($response['refresh_token']);
         }
 
-        $idToken = false;
-        if (isset($response['id_token'])) {
-            $idToken = $response['id_token'];
+        if (! empty($response['id_token'])) {
```

Comment:
Main bug fix is here ... only call `setIdToken()` (which validates) if there is one returned.

Reply:
I see you're using 3 different ways of checking if the value is set for the given properties before calling the associated "setProperty" method. 
- `if (isset($response['refresh_token']))`
- `if (! empty($response['id_token']))`
- `if ($user)`

Why is that?

Reply:
I kept `if (isset($response['refresh_token']))` because that's what it was before, no really good reason to change. This one won't error out attempting to set an empty one.

`if (! empty($response['id_token'])` is there because if it is empty then it will error out so need to be more careful. It's unlikely that it would be there but empty but still, better to be safe in this case. 

`if ($user)` needs to be not empty and not null. `empty($user)`, in this case would be the same.